nvbench/cuda_stream.cuh

```diff
@@ -28,19 +28,67 @@ namespace nvbench
 // RAII wrapper for a cudaStream_t.
 struct cuda_stream
 {
-  cuda_stream() { NVBENCH_CUDA_CALL(cudaStreamCreate(&m_stream)); }
-  ~cuda_stream() { NVBENCH_CUDA_CALL(cudaStreamDestroy(m_stream)); }
+  cuda_stream()
+      : m_owning{true}
+  {
+    NVBENCH_CUDA_CALL(cudaStreamCreate(&m_stream));
+  }
+
+  cuda_stream(cudaStream_t stream, bool owning)
+      : m_stream{stream}
+      , m_owning{owning}
+  {}
```

Comment:
```suggestion
  cuda_stream(cudaStream_t stream, bool owning = false)
      : m_stream{stream, destroy_stream{owning}}
  {}
```

Up to @allisonvacanti if this should default to non-owning and allow implicit conversion from `cudaStream_t`. 

May want to consider doing:
```
cuda_stream(int) = delete;
cuda_stream(std::nullptr_t) = delete;
```
to avoid ambiguity.

Reply:
I prefer to keep just the explicit two-parameter version. I've added a `nvbench::make_cuda_stream_view` helper function that can be used for convenience.